x/ibc-rate-limit/ibc_module.go

```diff
@@ -129,13 +129,18 @@ func (im *IBCModule) OnRecvPacket(
 	if err := ValidateReceiverAddress(packet); err != nil {
 		return osmoutils.NewEmitErrorAcknowledgement(ctx, types.ErrBadMessage, err.Error())
 	}
-
 	contract := im.ics4Middleware.GetContractAddress(ctx)
 	if contract == "" {
 		// The contract has not been configured. Continue as usual
 		return im.app.OnRecvPacket(ctx, packet, relayer)
 	}
-
+	// todo: move to OnChanOpenConfirm (however this requires figuring out how to extract a packet there)
```

Comment:
we probably want to figure out this before merging. on OnChanOpenConfirm we don't have access to the denom which may make this hard. An option might be to to add the rate limit with a channel "any" on denom creation